main.tf

```diff
@@ -1,13 +1,25 @@
 locals {
   is_t_instance_type = replace(var.instance_type, "/^t(2|3|3a){1}\\..*$/", "1") == "1" ? true : false
+  hostname_prefix    = var.hostname_prefix != "" ? var.hostname_prefix : "ip-`curl -s 169.254.169.254/latest/meta-data/local-ipv4/ | sed -e 's/\\./-/g'`"
```

Comment:
Just a question. what is this ?
```curl -s 169.254.169.254/latest/meta-data/local-ipv4/```

Reply:
Just get private ipaddress.
If `hostname_prefix` is "", `hostname` will set to `ip-${private_ipaddress}-${count}` .
```
[centos@bastion-1 ~]$ curl -s 169.254.169.254/latest/meta-data/local-ipv4/
10.42.0.251
```

Reply:
OK, I'm not sure if it is a common proper way. Is it ?

Reply:
Default `hostname` is set to `ip-xxx-xxx-xxx-xxx` by AWS.
But I did not remove the `count` suffix.

There may be better ways... 😓

Reply:
This seems like the way it is.
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html

Reply:
To get the hostname that is already assigned by AWS, `hostname --short` returns the same result as `curl ...`.
If there is any reason to get the AWS metadata from `169.254.169.254`, the following sounds like more straightforward.
`curl -s http://169.254.169.254/latest/meta-data/local-hostname | cut -d. -f1`
(`http://169.254.169.254/latest/meta-data/hostname` works the same way, I don't realize the difference though.)

Reply:
Thank! 
Since the hostname may change depending on the AMI or other reason, I will get IP to set the hostname securely.

Reply:
(Though I don't think it is very important because the point is just for a default value,) is it more secure to get an IP address than to get a hostname?

Reply:
>Though I don't think it is very important because the point is just for a default value

I agree with you.

>is it more secure to get an IP address than to get a hostname?

Yes, I think so. (I want to stop using the forked module 💦)